overlord/fdestate/fdestate.go

```diff
@@ -218,7 +220,14 @@ func initializeState(st *state.State) error {
 
 	// FIXME mount points will be different in recovery or factory-reset modes
 	// either inspect degraded.json, or use boot.HostUbuntuDataForMode()
-	dataUUID, dataErr := disksDMCryptUUIDFromMountPoint(dirs.SnapdStateDir(dirs.GlobalRootDir))
+
+	// On Classic, the data disk is mounted as /
+	dataDir := dirs.GlobalRootDir
+	if !release.OnClassic {
+		// If on Core /writable is a bind mount from data dir
+		dataDir = filepath.Join(dirs.GlobalRootDir, "writable")
+	}
```

Comment:
nitpick: it feels like this logic should be a function of the dirs package

Reply:
https://github.com/canonical/snapd/pull/14734/commits/457e84ef881d6dcd696171a3a0b8ab6a80e27075